include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -0,0 +1,246 @@
+
+    //!\brief The column over the optimal scores.
+    physical_column_t m_optimal_column{};
+    //!\brief The column over the horizontal gap scores.
+    physical_column_t m_horizontal_column{};
+    //!\brief The virtual column over the vertical gap scores.
+    virtual_column_t m_vertical_column{};
+    //!\brief The number of columns for this matrix.
+    size_t m_number_of_columns{};
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    score_matrix_single_column() = default; //!< Defaulted.
+    score_matrix_single_column(score_matrix_single_column const &) = default; //!< Defaulted.
+    score_matrix_single_column(score_matrix_single_column &&) = default; //!< Defaulted.
+    score_matrix_single_column & operator=(score_matrix_single_column const &) = default; //!< Defaulted.
+    score_matrix_single_column & operator=(score_matrix_single_column &&) = default; //!< Defaulted.
+    ~score_matrix_single_column() = default; //!< Defaulted.
+    //!\}
+
+    /*!\brief Resets the matrix and reallocates the memory of the underlying matrix.
+     * \tparam sequence1_t The type of the first sequence; must model std::ranges::forward_range.
+     * \tparam sequence2_t The type of the second sequence; must model std::ranges::forward_range.
+     *
+     * \param[in] sequence1 The first sequence.
+     * \param[in] sequence2 The second sequence.
+     */
+    template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
+    void reset_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
+    {
+        size_t number_of_rows = std::ranges::distance(sequence2) + 1;
+        m_optimal_column.resize(number_of_rows, 0);
+        m_horizontal_column.resize(number_of_rows, 0);
+        m_vertical_column = views::repeat_n(score_t{}, number_of_rows);
+        m_number_of_columns = std::ranges::distance(sequence1) + 1;
+    }
+
+    /*!\name Iterators
+     * \{
+     */
+    //!\brief Returns the iterator pointing to the first column.
+    matrix_iterator begin()
+    {
+        return matrix_iterator{*this};
+    }
+    //!\brief This score matrix is not const-iterable.
+    matrix_iterator begin() const = delete;
+
+    //!\brief Returns a default sentinel indicating the end of the matrix.
+    std::ranges::default_sentinel_t end()
+    {
+        return std::ranges::default_sentinel;
+    }
+
+    //!\brief This score matrix is not const-iterable.
+    std::ranges::default_sentinel_t end() const = delete;
+    //!\}
+};
+
+/*!\brief Score matrix for the pairwise alignment using only a single column.
+ * \implements std::input_iterator
+ *
+ * \details
+ *
+ * Implements a counted iterator to simulate the iteration over the actual matrix. When dereferencing the
+ * iterator returns a view over the allocated memory for the respective columns. The returned view is zips
+ * the three columns into a single range and transforms the returned tuple with the references to a
```

Comment:
```suggestion
 * the three columns into a single range and transforms the returned tuple to a
```

I know you want to describe `physical_column_t &`, etc. but I am not sure this is really important here - It's also hard to understand unless you see the code a few lines further down.